fili-core/src/main/java/com/yahoo/bard/webservice/web/util/PaginationParameters.java

```diff
@@ -99,15 +107,15 @@ private int parseParameter (String parameter, String parameterName) throws BadPa
     }
 
     /**
-     * Verifies that the passed parameter is greater than 0.
+     * Verifies that the passed parameter is greater than 0 or is -1 (last page).
      *
      * @param parameter  The parameter to be validated.
      * @param parameterName  The name of the parameter to appear in the error message
      *
-     * @throws BadPaginationException if 'parameter' is not greater than 0.
+     * @throws BadPaginationException if 'parameter' is not greater than 0 or is not -1.
      */
     private void validate(int parameter, String parameterName) throws BadPaginationException {
-        if (parameter < MINIMAL_VALUE) {
+        if (parameter < MINIMAL_VALUE && parameter != LAST_PAGE) {
```

Comment:
For sure but I would imagine this should still only apply to what `page` your asking for not the elements `perPage`